core_java/grep/src/main/java/ca/jrvs/apps/grep/javaGrepLambda.java

```diff
@@ -0,0 +1,59 @@
+package ca.jrvs.apps.grep;
+
+import static java.util.regex.Pattern.compile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class javaGrepLambda extends javaGrepImp {
+
+  @Override
+  public List<File> listFiles(String rootDir) {
+    List<File> returnList = new ArrayList<File>();
+    try {
+      returnList = Files.walk(Paths.get(rootDir))
+          .map(x-> x.toFile())
+          .filter(x -> x.isFile())
```

Comment:
You should be able to use (File::isFile) and (Files::toFile) here. IntelliJ will tell you when you can do that.